### DISCUSSION TITLE
How to inject command options to provider

### DISCUSSION THREAD

**Discussion by hannoeru** [jmcdo29/nest-commander#729] [Q&A]
Can I get or inject command options inside provider?

**Comment by jmcdo29** [comment]
[You can get the current command via `this.command` in the command class.](https://nest-commander.jaymcdoniel.dev/en/features/commander/#getting-the-current-command) No need to inject it, it's available inside of the `run` method or any other user defined method. From the command instance you should be able to get the options you are looking for

**Reply by hannoeru** [reply]
I am developing a multi-tenant application that using custom provider for database connection, and using this package to build the seeder for database.

The original custom provider is request scoped that resolve which tenant should be used:
```ts
export const TENANT_CONTEXT = Symbol('TENANT_CONTEXT');
export const TENANT_CONNECTION = Symbol('TENANT_CONNECTION');

export const TenantContextProvider: Provider = {
  provide: TENANT_CONTEXT,
  scope: Scope.REQUEST,
  inject: [REQUEST, TenantsService],
  useFactory: async (
    req: FastifyRequest<RequestGeneric>,
    tenantService: TenantsService,
  ): Promise<Tenant> => {
    const tenantId = await getTenantIdFromRequest(req);
    if (!tenantId) {
      throw new NotFoundException('Tenant not found');
    }
    const tenant = await tenantService.findByName(tenantId);
    if (!tenant) {
      throw new NotFoundException('Tenant not found');
    }
    return tenant;
  },
};

export const TenantConnectionProvider: Provider = {
  provide: TENANT_CONNECTION,
  inject: [TENANT_CONTEXT, TenantConnection],
  useFactory: async (
    context: Tenant,
    connection: TenantConnection,
  ): Promise<mongoose.Connection> => {
    // Set tenant context
    connection.tenantId = context.name;

    // Return the connection
    return connection.getConnection(context);
  },
};
```

Because there is no request in commander, so I replaced it with this provider.
But I can't figure out how to get tenant from command options here
```ts
export const SeederTenantContextProvider: Provider = {
  provide: TENANT_CONTEXT,
  inject: [ConfigService],
  useFactory: (configService: ConfigService): Tenant => {
    return {
      name: 'dev', // How can I get the command options here
      dbUri: configService.get('database.master.uri')!,
    };
  },
};
```

Thanks for your quick reply.

**Reply by jmcdo29** [reply]
Ah, this is an option passed via the cli flags yeah?

**Reply by hannoeru** [reply]
Yes
```ts
@Option({
  flags: '-t, --tenant [string]',
  required: true,
})
parseTenant(value: string): string {
  return value;
}
```

**Reply by jmcdo29** [reply]
Well, I _was_ going to say you could get the commander instance, but it turns out that I don't have that token exposed, only a decorator, which isn't viable in the `inject` array, so I'll need to make a patch for that. Once I do, you'll be able to inject the entire commander instance and use the `commander.opts()` method to get the options. However, this might be before commander has parsed the options, and if that's the case, you'll be better off reading from `process.argv`.

This is an interesting case. I'll keep thinking on what could be done to help this if anything

**Reply by hannoeru** [reply]
Reading from `process.argv` works. Thank you for your help!
